### PR TITLE
Create initial codeql.yml configuration

### PR DESCRIPTION
**Pull request by jentfoo**
This PR adds a new CodeQL configuration.

### CODE REVIEW

**Review by adaadb6 — APPROVED** [review]

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+on:
+  push:
+    branches:
+      - master
+      - branch/*
+  pull_request:
+    branches:
+      - master
+      - branch/*
```
Comment (reedloden) [inline .github/workflows/codeql.yml:11]: See above.

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches:
+      - master
+      - branch/*
```
Comment (reedloden) [inline .github/workflows/codeql.yml:7]: The branch names on this repository are different from `teleport`. It's actually `teleport-v*`.

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
+      with:
+        languages: ${{ matrix.language }}
+
+    # - run: |
```
Comment (reedloden) [inline .github/workflows/codeql.yml:39]: Can we remove this? Do we need to include autobuild at all?

File: .github/workflows/codeql.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
+name: "CodeQL"
```
Comment (reedloden) [inline .github/workflows/codeql.yml:1]: We use `codeql-analysis.yml` on other repositories. I don't have a strong preference as to which to use, but should standardize, imho.

**Review by reedloden — COMMENTED** [review]
This replaces https://github.com/gravitational/webapps/pull/1103.

This will need to be backported to all supported branches.

File: .github/workflows/codeql.yml (hunk 4 above)
Comment (jentfoo) [reply .github/workflows/codeql.yml:1]: Which standard would be easier to adopt?  GitHub choose the `codeql.yml` name as a default

**Comment by jentfoo** [comment]
I will address these recommendations shortly, thank you for the comments @reedloden!

File: .github/workflows/codeql.yml (hunk 4 above)
Comment (jentfoo) [reply .github/workflows/codeql.yml:1]: I only see three cases of `codeql-analysis.yml`.  If it sounds good I suggest we transition to the default naming to reduce the manual steps of on-boarding a project.

I can submit PR's for the 3 projects we have today

**Pull request merged**